Automattic/pocket-casts-android
# Fix edge-to-edge for the full-screen player

## Description

This change focuses on converting the full-screen to edge-to-edge for the Android 15 project. Please ignore the other pages for now. 

There will be multiple pull requests required to support Android 15 edge-to-edge but I'm going to break them down into smaller PRs. I will merge them all into a `task/android-15` branch, and after all the work is completed, raise a PR to merge into `main`.

## Testing Instructions

Open the following pages and verify the content isn't under the status or navigation bar.

1. Subscribe to Stuff You Should Know (a podcast with transcripts)
2. Open the full-screen page
3. Tap on the effects icon
4. Close and tap on the sleep icon
5. Close and tap on the rearrange three dots
6. Tap on the pencil to edit the actions
7. Go back to the full-screen player
8. Tap on the Up Next icon
9. Go back and open the transcripts page

## Screenshots

### Now Playing tab

**Before**

<img width="1479" alt="Screenshot 2024-12-05 at 2 14 20 pm" src="https://github.com/user-attachments/assets/b3dde640-64b6-47e8-ab6b-1378eea67d96">

**After**

<img width="1477" alt="Screenshot 2024-12-05 at 1 38 50 pm" src="https://github.com/user-attachments/assets/fa8d3eca-72b3-4800-9ac9-1c4034d62bfb">

### Up Next

**Before**

<img width="1479" alt="Screenshot 2024-12-05 at 2 15 41 pm" src="https://github.com/user-attachments/assets/518b1797-ceea-4f32-a92a-a0bdd37159a0">

**After**

<img width="1478" alt="Screenshot 2024-12-05 at 1 40 31 pm" src="https://github.com/user-attachments/assets/160c18b4-42e0-45a0-9cdd-019e75722217">

### Effects 

**Before**

<img width="1480" alt="Screenshot 2024-12-05 at 2 17 50 pm" src="https://github.com/user-attachments/assets/d69a7c5a-8572-4ce6-9195-532e107e0556">

**After**

The navigation bar on SDK 35 isn't quite right for the bottom sheet but I'm going to leave this for later.

<img width="1478" alt="Screenshot 2024-12-05 at 10 17 00 pm" src="https://github.com/user-attachments/assets/41060bf7-8780-475e-ba5f-91d5f06852e5">

### Sleep 

**Before**

<img width="1477" alt="Screenshot 2024-12-05 at 2 18 02 pm" src="https://github.com/user-attachments/assets/123e8921-3b85-4208-8867-7e9951f02239">

**After**

<img width="1480" alt="Screenshot 2024-12-05 at 10 17 14 pm" src="https://github.com/user-attachments/assets/458064b5-34f6-43b0-80d7-350cee6ce2b0">

### More actions 

**Before**

<img width="1478" alt="Screenshot 2024-12-05 at 2 20 32 pm" src="https://github.com/user-attachments/assets/2c40fc70-2c8b-4771-aa22-3c6a0c4ccddb">

**After**

<img width="1478" alt="Screenshot 2024-12-05 at 10 19 07 pm" src="https://github.com/user-attachments/assets/19a1dd98-417f-482f-bec2-4fe1b7c9b8f8">

### Rearrange actions

**Before**

<img width="1475" alt="Screenshot 2024-12-05 at 2 19 34 pm" src="https://github.com/user-attachments/assets/d461f470-beed-436a-b53d-f2d3c1f598db">

**After**

<img width="1478" alt="Screenshot 2024-12-05 at 3 41 41 pm" src="https://github.com/user-attachments/assets/1e322757-836f-4fc2-aee5-cbd9594d545e">

### Transcripts

**Before**

<img width="1478" alt="Screenshot 2024-12-05 at 3 56 11 pm" src="https://github.com/user-attachments/assets/553c8650-521d-41f4-aeb6-10efa6b5fc5d">

**After**

<img width="1481" alt="Screenshot 2024-12-05 at 3 51 24 pm" src="https://github.com/user-attachments/assets/00b858ae-aec8-49f9-a188-b2a55270d0c7">


## Checklist
 
#### I have tested any UI changes...
